packages/go_router/lib/src/delegate.dart

```diff
@@ -167,6 +168,10 @@ class GoRouterDelegate extends RouterDelegate<RouteMatchList>
     }());
   }
 
+  /// The current [GoRouterState]
+  GoRouterState? get currentState => currentConfiguration.last
+      .buildState(_configuration, currentConfiguration);
```

Comment:
Is there a way to not rebuild the state here ?

Reply:
no unless you caches in the `GoRouterStateRegistry`, which is something i was planning to do at some point. 

but I won't worry too much about this at the moment.

Reply:
also the document should mention what will be returned if you use GoRouter.push()